src/sentry/models/project.py

```diff
@@ -224,15 +227,19 @@ def is_internal_project(self):
                 return True
         return False
 
-    # TODO: Make these a mixin
-    def update_option(self, *args, **kwargs):
-        return projectoptions.set(self, *args, **kwargs)
+    @property
+    def option_manager(self) -> OptionManager:
+        from sentry.models import ProjectOption
+
+        return ProjectOption.objects
 
-    def get_option(self, *args, **kwargs):
-        return projectoptions.get(self, *args, **kwargs)
+    def update_option(self, key: str, value: Value) -> None:
+        projectoptions.update_rev_for_option(self)
+        return super().update_option(key, value)
 
-    def delete_option(self, *args, **kwargs):
-        return projectoptions.delete(self, *args, **kwargs)
+    def delete_option(self, key: str) -> None:
+        projectoptions.update_rev_for_option(self)
```

Comment:
Longer term should we align and combine the `projectoptions` wrappers with the model APIs? Continuing to have multiple paths to set project options seems like a trap for future us.

Reply:
I agree.